src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -670,9 +670,15 @@ export function registerTerminalActions() {
 			const labelService = accessor.get(ILabelService);
 			const remoteAgentService = accessor.get(IRemoteAgentService);
 			const notificationService = accessor.get(INotificationService);
-			const offProcTerminalService = remoteAgentService.getConnection() ? accessor.get(IRemoteTerminalService) : accessor.get(ILocalTerminalService);
-			const remoteTerms = await offProcTerminalService.listProcesses();
-			const unattachedTerms = remoteTerms.filter(term => !terminalService.isAttachedToTerminal(term));
+			let offProcTerminalService = remoteAgentService.getConnection() ? accessor.get(IRemoteTerminalService) : accessor.get(ILocalTerminalService);
+
+			const terms = await offProcTerminalService.listProcesses();
+
+			if ('reduceConnectionGraceTime' in offProcTerminalService) {
+				offProcTerminalService.reduceConnectionGraceTime();
+			}
```

Comment:
Thinking about this more, let's do this in local too so they work the same. It shouldn't hurt to reduce here since when attach is run